server.js

```diff
@@ -1,35 +1,215 @@
-import express from "express";
-import cors from "cors";
-import mongoose from "mongoose";
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
-mongoose.connect(mongoUrl);
-mongoose.Promise = Promise;
+import express from "express"
+import cors from "cors"
+import mongoose from "mongoose"
+import expressListEndpoints from "express-list-endpoints"
+import { Restaurant } from "./models/Restaurant"
+import michelinData from "./data/michelin-data.json"
+
+//Set up MongoURL and localhost
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/restaurants"
+mongoose.connect(mongoUrl)
+mongoose.Promise = Promise
+
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Restaurant.deleteMany()
+    //insert each document in the array into the collection
+    await Restaurant.insertMany(michelinData)
+  }
+  seedDatabase()
+}
 
 // Defines the port the app will run on. Defaults to 8080, but can be overridden
 // when starting the server. Example command to overwrite PORT env variable value:
 // PORT=9000 npm start
-const port = process.env.PORT || 8080;
-const app = express();
+const port = process.env.PORT || 3000
+const app = express()
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors());
-app.use(express.json());
+app.use(cors())
+app.use(express.json())
 
-// Start defining your routes here
+// Endpoint listing and guidance for using query endpoints
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
-});
+  const endpoints = expressListEndpoints(app).map((endpoint) => {
```

Comment:
Ah this is how you added all the descriptions. I was wondering. Great thinking that you explain them here like that in your API! 💯